### PR TITLE
clean up icon path confusion

### LINKED ISSUE #35491: clean up icon path confusion
With https://github.com/Microsoft/vscode/commit/269ae60a4b7f0afb4d779433d88d9c1ef2190bc6 the concept of an `iconPath` has been added. This competes with `iconClass` and having both doesn't make much sense. While I understand the reasoning for us having it `iconPath` is only honoured in a single place and every other user will fail to make it work. At least the `MenuItemActionItem` should know what to do, ideally no competing concepts exist.

### PR DESCRIPTION
**Pull request by bpasero**
fixes #35491

### CODE REVIEW

**Comment by bpasero** [comment]
@jrieken ping

**Comment by jrieken** [comment]
ops, forgot to press submit...

File: src/vs/platform/actions/browser/menuItemActionItem.ts [hunk 1]
```diff
@@ -114,17 +117,22 @@ export function createActionItem(action: IAction, keybindingService: IKeybinding
...
 }
 
+const ids = new IdGenerator('menu-item-action-item-icon-');
+
 export class MenuItemActionItem extends ActionItem {
 
+	static readonly ICON_PATH_TO_CSS_RULES: Map<string /* path*/, string /* CSS rule */> = new Map<string, string>();
+
 	private _wantsAltCommand: boolean = false;
+	private _itemClassDispose: IDisposable;
```
[the remark this answers is not in the document]
Comment (bpasero) [reply src/vs/platform/actions/browser/menuItemActionItem.ts]: @jrieken same as before I would say (https://github.com/Microsoft/vscode/pull/41577/files#diff-b1eab124b1401b8228c34df2fc86a5e7L303) ?

File: src/vs/platform/actions/browser/menuItemActionItem.ts [hunk 2]
```diff
@@ -114,17 +117,22 @@ export function createActionItem(action: IAction, keybindingService: IKeybinding
...
 	constructor(
-		action: MenuItemAction,
+		private action: MenuItemAction,
 		@IKeybindingService private _keybindingService: IKeybindingService,
 		@IMessageService protected _messageService: IMessageService,
 		@IContextMenuService private _contextMenuService: IContextMenuService
 	) {
-		super(undefined, action, { icon: !!action.class, label: !action.class });
+		super(undefined, action, { icon: !!(action.class || action.item.iconPath), label: !action.class && !action.item.iconPath });
 	}
```
[the remark this answers is not in the document]
Comment (bpasero) [reply src/vs/platform/actions/browser/menuItemActionItem.ts]: @jrieken this is just a hint that we have an icon to show

File: src/vs/platform/actions/common/actions.ts [hunk 3]
```diff
@@ -23,8 +23,7 @@ export interface ICommandAction {
 	id: string;
 	title: string | ILocalizedString;
 	category?: string | ILocalizedString;
-	iconClass?: string;
-	iconPath?: string;
+	iconPath?: string | { light: string; dark: string; };
```
[the remark this answers is not in the document]
Comment (bpasero) [reply src/vs/platform/actions/common/actions.ts]: @jrieken yeah makes sense, pushed

File: src/vs/platform/actions/browser/menuItemActionItem.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrieken) [reply src/vs/platform/actions/browser/menuItemActionItem.ts]: Interesting line of argumentation... Well, now that this is created by an object with a lifecycle we could make use of that fact...

File: src/vs/platform/actions/browser/menuItemActionItem.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrieken) [reply src/vs/platform/actions/browser/menuItemActionItem.ts]: and that hint is used somewhere?

[the remark this answers is not in the document]
Comment (bpasero) [reply src/vs/platform/actions/browser/menuItemActionItem.ts]: @jrieken here to add the icon class I guess: https://github.com/Microsoft/vscode/blob/ben/fix-35491/src/vs/base/browser/ui/actionbar/actionbar.ts#L302

File: src/vs/platform/actions/browser/menuItemActionItem.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bpasero) [reply src/vs/platform/actions/browser/menuItemActionItem.ts]: @jrieken sorry I thought you meant something else, pushed a fix

File: src/vs/platform/actions/browser/menuItemActionItem.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrieken) [reply src/vs/platform/actions/browser/menuItemActionItem.ts]: I see

**Comment by jrieken** [comment]
lgtm

**Pull request merged**